java/ql/test/utils/model-generator/typebasedflow/p/Stream.java

```diff
@@ -0,0 +1,249 @@
+package p;
+
+import java.util.*;
+import java.util.function.*;
+import java.util.stream.LongStream;
+import java.util.stream.IntStream;
+import java.util.stream.DoubleStream;
+import java.util.stream.Collector;
+
+/**
+ * This is a stub implementation of the Java Stream API.
+ */
+public class Stream<T> {
+
+    // MaD=p;Stream;true;iterator;();;Argument[-1].Element;ReturnValue.Element;value;generated
+    public Iterator<T> iterator() {
+        return null;
+    }
+
+    // MaD=p;Stream;true;allMatch;(Predicate);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public boolean allMatch(Predicate<? super T> predicate) {
+        return false;
+    }
+
+    // MaD=p;Stream;true;collect;(Supplier,BiConsumer,BiConsumer);;Argument[-1].Element;Argument[1].Parameter[1];value;generated
+    // MaD=p;Stream;true;collect;(Supplier,BiConsumer,BiConsumer);;Argument[0].ReturnValue;Argument[1].Parameter[0];value;generated
+    // MaD=p;Stream;true;collect;(Supplier,BiConsumer,BiConsumer);;Argument[0].ReturnValue;Argument[2].Parameter[0];value;generated
+    // MaD=p;Stream;true;collect;(Supplier,BiConsumer,BiConsumer);;Argument[0].ReturnValue;Argument[2].Parameter[1];value;generated
+    // MaD=p;Stream;true;collect;(Supplier,BiConsumer,BiConsumer);;Argument[0].ReturnValue;ReturnValue;value;generated
+    public <R> R collect(Supplier<R> supplier, BiConsumer<R, ? super T> accumulator, BiConsumer<R, R> combiner) {
+        return null;
+    }
+
+    // Collector is not a functional interface, so this is not supported
+    public <R, A> R collect(Collector<? super T, A, R> collector) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;concat;(Stream,Stream);;Argument[0].Element;ReturnValue.Element;value;generated
+    // MaD=p;Stream;true;concat;(Stream,Stream);;Argument[1].Element;ReturnValue.Element;value;generated
+    public static <T> Stream<T> concat(Stream<? extends T> a, Stream<? extends T> b) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;distinct;();;Argument[-1].Element;ReturnValue.Element;value;generated
+    public Stream<T> distinct() {
+        return null;
+    }
+
+    public static <T> Stream<T> empty() {
+        return null;
+    }
+
+    // MaD=p;Stream;true;filter;(Predicate);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    // MaD=p;Stream;true;filter;(Predicate);;Argument[-1].Element;ReturnValue.Element;value;generated
+    public Stream<T> filter(Predicate<? super T> predicate) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;findAny;();;Argument[-1].Element;ReturnValue.SyntheticField[ArgType0];value;generated
+    public Optional<T> findAny() {
+        return null;
+    }
+
+    // MaD=p;Stream;true;findFirst;();;Argument[-1].Element;ReturnValue.SyntheticField[ArgType0];value;generated
+    public Optional<T> findFirst() {
+        return null;
+    }
+
+    // public <R> Stream<R> flatMap(Function<? super T, ? extends Stream<? extends
```

Comment:
Why is this test case excluded?

Reply:
If it doesn't work, then we might as well keep it with a comment about why and what doesn't work.

Reply:
That is an oversight on my part - it does work and I will comment it back in.